src/main/kotlin/org/rust/ide/typing/assist/FunctionOrStructFixer.kt

```diff
@@ -0,0 +1,57 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.typing.assist
+
+import com.intellij.lang.SmartEnterProcessorWithFixers
+import com.intellij.openapi.editor.Editor
+import com.intellij.psi.PsiElement
+import org.rust.lang.core.psi.RsElementTypes.*
+import org.rust.lang.core.psi.RsFunction
+import org.rust.lang.core.psi.RsStructItem
+import org.rust.lang.core.psi.ext.body
+import org.rust.lang.core.psi.ext.elementType
+import org.rust.lang.core.psi.ext.endOffset
+
+class FunctionOrStructFixer : SmartEnterProcessorWithFixers.Fixer<RsSmartEnterProcessor>() {
+
+    override fun apply(editor: Editor, processor: RsSmartEnterProcessor, element: PsiElement) {
+        if (element.elementType !in APPROVED_TYPES) return
+
+        val parent = element.parent
+        val content = when {
+            parent.isFnAndIdentifier() -> "() {\n\n}"
```

Comment:
Maybe adding tabs is better? But `\t` may not be suitable.

Reply:
I think `\n` is ok.